### ISSUE TITLE
Stalling detection scheme

### ISSUE THREAD

**Issue by Wercal** [ClubSynapsETS/Prothese#2]
The currently designed stalling detection scheme for the finger actuators is a simple voltage read on each ADC channel a frequency of 10 Hz and comparing with the last three reads when the current motor the finger state. This only routine only occurs when the finger is in movement. This translate in C as: if(FG_CLOSING || FG_OPENING) : xSemphoreGive(StallingDetectSmphr) //unblock the stalling detection task.

Whenever a stall is detected, a very similar voltage is detected three times in a row, an event is called in the finger_interface task and stops the finger and changes it's state to either FG_STALLING_CLOSING or FG_STALLING_OPENING. 

NOTE: The solution to issue **ADC timing defined and integrated** should also be applied to reduce stalling detection time. The stalling detection scheme is not part of the finger interface time slot but should still take a little time as possible.

**Comment by mastodonde** [comment]
Mesurer du courant lorsqu'il est a la fin de sa source (spike courant ).
